### ISSUE TITLE
`Style/MapToHash` incorrectly flags non-hash transformations

### ISSUE THREAD

**Issue by ojab** [rubocop/rubocop#10314]
```ruby
%w[path other_path].map { |path| Pathname.new(path) }.to_h { |path| [path.basename, path.extname] }
```
leads to
```
Offenses:

/tmp/1.rb:1:21: C: Style/MapToHash: Pass a block to to_h instead of calling map.to_h.
%w[path other_path].map { |path| Pathname.new(path) }.to_h { |path| [path.basename, path.extname] }
```
despite the fact that the `#map` is not creating a Hash.  

## RuboCop version
`1.24.0 (using Parser 3.0.3.2, rubocop-ast 1.15.0, running on ruby 3.0.3 x86_64-linux)``

**Comment by tejasbubane** [comment]
Not sure if this is a bug. Same can be rewritten as following to avoid offense:

```ruby
%w[path other_path].to_h do |path| 
  pathname = Pathname.new(path)
  [pathname.basename, pathname.extname]
end
```

We have tests to specifically check for this case `map <block> to_h <block>` - and register offense but do not autocorrect - which seems correct.
https://github.com/rubocop/rubocop/blob/9f39dad99d6d4cdc2c1f96e205000ca081a820ce/spec/rubocop/cop/style/map_to_hash_spec.rb#L79-L88

**Comment by dvandersluis** [comment]
Yeah it’s not a bug IMO as @tejasbubane mentioned it can be rewritten without the `map`.

**Comment by ojab** [comment]
Yep, but unfortunately it clashes with `Avoid do…​end when chaining.` if there are further method calls.
```ruby
CONST = %w[path other_path].to_h do |path| 
  pathname = Pathname.new(path)
  [pathname.basename, pathname.extname]
end.freeze
```
wouldn't fly.
